dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/fileresource/FileResourceCleanUpJob.java

```diff
@@ -70,125 +106,108 @@ public JobType getJobType() {
-
-    // Delete unassigned JOB_DATA file resources that have no associated job config
-    List<FileResource> unassignedJobDataFileResources =
+    // Job Data FRs
+    List<FileResource> jobDataUnassigned =
         fileResourceService.getAllUnassignedByJobDataDomainWithNoJobConfig();
-    progress.startingStage(
-        "Deleting JOB_DATA file resources associated with deleted ONCE_ASAP jobs",
-        unassignedJobDataFileResources.size(),
-        SKIP_ITEM_OUTLIER);
+    deleteFrs(
+        progress,
+        "Deleting unassigned JOB_DATA file resources associated with deleted ONCE_ASAP jobs",
+        jobDataUnassigned,
+        deletedFileResources);
+
+    // Remaining applicable FRs to be deleted
+    List<FileResource> remainingUnassigned =
+        fileResourceService.getExpiredFileResources(domainsToDeleteWhenUnassigned);
+    deleteFrs(
+        progress,
+        "Deleting unassigned file resources for domains %s"
+            .formatted(
+                domainsToDeleteWhenUnassigned.stream()
+                    .map(Enum::name)
+                    .collect(Collectors.joining())),
+        remainingUnassigned,
+        deletedFileResources);
+
+    log.info(
+        "Deleted {} unassigned FileResources {}",
+        deletedFileResources.size(),
+        deletedFileResources.isEmpty() ? "" : deletedFileResources);
+
+    progress.completedProcess("File resource clean-up complete");
+  }
+
+  private void deleteFrs(
+      JobProgress progress,
+      String message,
+      List<FileResource> fileResources,
+      List<UnassignedFileResource> deletedFileResources) {
+    progress.startingStage(message, fileResources.size(), SKIP_ITEM_OUTLIER);
     progress.runStage(
-        unassignedJobDataFileResources,
+        fileResources,
         FileResourceCleanUpJob::toIdentifier,
         fr -> {
           if (safeDelete(fr)) {
-            deletedFileResourcesForDeletedJobs.add(new SimpleEntry<>(fr.getName(), fr.getUid()));
+            deletedFileResources.add(
+                new UnassignedFileResource(fr.getUid(), fr.getName(), fr.getDomain().name()));
           }
         });
-
-    if (!deletedOrphans.isEmpty()) {
-      log.info(
-          String.format(
-              "Deleted %d orphaned FileResources: %s",
-              deletedOrphans.size(), prettyPrint(deletedOrphans)));
-    }
-
-    if (!deletedExpired.isEmpty()) {
-      log.info(
-          String.format(
-              "Deleted %d expired data value FileResources: %s",
-              deletedExpired.size(), prettyPrint(deletedExpired)));
-    }
-
-    if (!deletedFileResourcesForDeletedJobs.isEmpty()) {
-      log.info(
-          String.format(
-              "Deleted %d FileResource(s) for deleted jobs : %s",
-              deletedFileResourcesForDeletedJobs.size(),
-              prettyPrint(deletedFileResourcesForDeletedJobs)));
-    }
-    progress.completedProcess(null);
-    log.info("Finished FileResourceCleanUpJob");
+    progress.completedStage(null);
```

Comment:
`runStage` includes either calling `completedStaged` or `failedStage` so we should not call it explicitly here